image/verify/src/verifier.rs

```diff
@@ -216,24 +221,32 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
 
         match key_idx {
             0..=SECOND_LAST_KEY_IDX => {
+                cfi_assert_le(cfi_launder(key_idx), SECOND_LAST_KEY_IDX);
                 let key = VendorPubKeyRevocation::from_bits_truncate(0x01u32 << key_idx);
-                if revocation.contains(key) {
+                if cfi_launder(revocation.contains(key)) {
                     Err(CaliptraError::IMAGE_VERIFIER_ERR_VENDOR_ECC_PUB_KEY_REVOKED)?;
+                } else {
+                    cfi_assert!(!revocation.contains(key));
```

Comment:
Need to double check the assembly that this re-does the full contains operations instead of just recomparing the result (which is what I suspect).

Reply:
Yeah, revocation should be laundered. Hardened it even further by laundering the key as well.

![image](https://github.com/chipsalliance/caliptra-sw/assets/38512878/86776926-0c7c-407d-8d84-ca81b6274915)